src/modules/battery.cpp

```diff
@@ -156,6 +174,47 @@ const std::tuple<uint8_t, float, std::string, float> waybar::modules::Battery::g
   std::lock_guard<std::mutex> guard(battery_list_mutex_);
 
   try {
+#if defined(__FreeBSD__)
+    int capacity;
+    size_t size_capacity = sizeof capacity;
+    if (sysctlbyname("hw.acpi.battery.life", &capacity, &size_capacity, NULL,0) != 0) {
+      throw std::runtime_error("sysctl hw.acpi.battery.life failed");
+    }
+    int time;
+    size_t size_time = sizeof time;
+    if (sysctlbyname("hw.acpi.battery.time", &time, &size_time, NULL,0) != 0) {
+      throw std::runtime_error("sysctl hw.acpi.battery.time failed");
+    }
+    int rate;
+    size_t size_rate = sizeof rate;
+    if (sysctlbyname("hw.acpi.battery.rate", &rate, &size_rate, NULL,0) != 0) {
+      throw std::runtime_error("sysctl hw.acpi.battery.rate failed");
+    }
+
+    auto status = getAdapterStatus(capacity);
+    // Handle full-at
+    if (config_["full-at"].isUInt()) {
+      auto full_at = config_["full-at"].asUInt();
+      if (full_at < 100) {
+        capacity = 100.f * capacity / full_at;
+      }
+    }
+    if (capacity > 100.f) {
+      // This can happen when the battery is calibrating and goes above 100%
+      // Handle it gracefully by clamping at 100%
+      capacity = 100.f;
+    }
+    uint8_t cap = round(capacity);
+    if (cap == 100 && status=="Plugged") {
+      // If we've reached 100% just mark as full as some batteries can stay
+      // stuck reporting they're still charging but not yet done
+      status = "Full";
+    }
+
+    //spdlog::info("{} {} {} {}", capacity,time,status,rate);
+    return {capacity, time, status, rate};
+
+#else
```

Comment:
Cannot use `#else` here when `inotify_init1()` is hidden behind `#if defined(__Linux__)`.
```suggestion
#elif defined(__linux__)
```